File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {
    int square_of_sum(int n) {
        int s=0;
        for (int i=1; i<=n; i++) {
            s+=i;
        }
        return s*s;
    }
    int sum_of_squares(int n) {
        int s=0;
        for (int i=1; i<=n; i++) {
            s+=i*i;
        }
        return s;
    }
    int difference(int n) {
        return square_of_sum(n) - sum_of_squares(n);
    }
}  // namespace difference_of_squares

File: solutions/fortran/difference-of-squares/1/difference_of_squares.f90
module difference_of_squares
  implicit none
contains
  integer function square_of_sum(n)
  integer :: n, i
  square_of_sum=0
  do i = 1,n 
    square_of_sum=square_of_sum+i
  end do
  square_of_sum = square_of_sum * square_of_sum
  return
  end function square_of_sum

  integer function sum_of_squares(n)
  integer :: n, i
  sum_of_squares=0
  do i = 1,n 
    sum_of_squares=sum_of_squares+i*i
  end do
  return
  end function sum_of_squares

  integer function difference(n)
    integer :: n
    difference = square_of_sum(n) - sum_of_squares(n)
    return
  end function difference

end module difference_of_squares
